.github/workflows/lint.yml

```diff
@@ -45,19 +44,13 @@ jobs:
     - uses: actions/checkout@v4
       with:
         persist-credentials: false
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: "3"
     - name: Install uv
       uses: astral-sh/setup-uv@v6
       with:
         version: latest
         enable-cache: false
-    - name: Install dependencies
-      run: uv pip install -r pyproject.toml --group package --group test --group types
     - name: Type check with mypy
-      run: mypy
+      run: uv run mypy
```

Comment:
Which packages does `uv` install here? Does it pick up the dependency groups automatically?

Reply:
This is skipping the fine-grained dependency groups. This could be readded but the micro optimisation just didn't seem worth the complexity

Reply:
in fact, given UV resolves the dependencies damn near instantly, i'd be in favour of collapsing these groups into something much more simple